index.js

```diff
@@ -18,19 +19,20 @@ function useCustom(store, React, mapState, mapActions) {
     const newListener = { oldState: {} };
     newListener.run = mapState
       ? newState => {
-          const mappedState = mapState(newState);
-          if (mappedState !== newListener.oldState) {
-            newListener.oldState = mappedState;
-            originalHook(mappedState);
-          }
+        const mappedState = mapState(newState);
+        if (mappedState !== newListener.oldState) {
+          newListener.oldState = mappedState;
+          originalHook(mappedState);
         }
+      }
       : originalHook;
```

Comment:
originalHook needs to be called with a new reference each time, to ensure React re-renders
- see https://github.com/andregardi/use-global-hook/pull/59